### DISCUSSION TITLE
Difference between `@prisma/client` and `@prisma/client/edge`

### DISCUSSION THREAD

**Discussion by wfjsw** [prisma/prisma#24407] [Q&A] [answered]
### Question

What is the difference between

```
import { PrismaClient } from '@prisma/client'
```
and 
```
import { PrismaClient } from '@prisma/client/edge'
```
? 

I've seen them interchangably in the issues and docs. Assuming I am using Cloudflare D1.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @wfjsw 👋 

Thank you for raising this question.

The difference between `import { PrismaClient } from '@prisma/client'` and `import { PrismaClient } from '@prisma/client/edge'` lies in their intended use cases and environments:

1. **`@prisma/client`**:
   - This is the standard import for Prisma Client, used in typical server-side environments.
   - It is suitable for traditional server environments like Node.js where you don't have the constraints of edge computing.

   ```javascript
   import { PrismaClient } from '@prisma/client'
   
   const prisma = new PrismaClient()
   // use `prisma` in your application to read and write data in your DB
   ```

2. **`@prisma/client/edge`**:
   - This import is specifically designed for edge environments, such as Cloudflare Workers.
   - Edge environments have different performance and resource constraints compared to traditional server environments, and `@prisma/client/edge` is optimized to handle these constraints.

   ```javascript
   import { PrismaClient } from '@prisma/client/edge'
   
   const prisma = new PrismaClient()
   // use `prisma` in your application to read and write data in your DB
   ```

For your specific case of using Cloudflare D1, you should use the `@prisma/client/edge` import to ensure compatibility with the edge runtime. Here is an example of how you can set it up:

```typescript
import { PrismaClient } from '@prisma/client/edge'
import { PrismaD1 } from '@prisma/adapter-d1'

export interface Env {
  DB: D1Database
}

export default {
  async fetch(
    request: Request,
    env: Env,
    ctx: ExecutionContext
  ): Promise<Response> {
    const adapter = new PrismaD1(env.DB)
    const prisma = new PrismaClient({ adapter })

    const users = await prisma.user.findMany()
    const result = JSON.stringify(users)
    return new Response(result)
  },
}
```

This setup ensures that your Prisma Client is configured correctly for the Cloudflare Workers environment.

For more detailed information, you can refer to the [documentation on deploying to Cloudflare Workers](https://www.prisma.io/docs/orm/prisma-client/deployment/edge/deploy-to-cloudflare#5-use-prisma-client-in-your-worker-to-send-a-query-to-the-database-2).

**Reply by wfjsw** [reply]
Thank you! That makes much more sense. The bundle size goes down to 10% after I replaced `@prisma/client` to `@prisma/client/edge`. 

The documentation is still mentioning `@prisma/client`. Probably worth changing that?

**Reply by wfjsw** [reply]
My client stops working after making changes according to this:

```
(error) PrismaClientValidationError: Prisma Client was configured to use the `adapter` option but it was imported via its `/edge` endpoint.
Please either remove the `/edge` endpoint or remove the `adapter` from the Prisma Client constructor.
```

**Reply by janpio** [reply, marked as answer]
`/edge` is not a thing when you want to use Driver Adapters via the `driverAdapters` preview feature, unfortunately. That is also why the D1 documentation does not mention this but just normal `@prisma/client`.

**Reply by NomadicDeveloper22** [reply]
So it's not possible to use /edge on Cloudflare pages, while trying to use Neon as a database?

**Comment by wc-ab** [comment]
What about Vercel Function on nodeJS runtime (Serverless Functions)? Should I use the edge client or the normal one?
